pkg/registry/generic/rest/proxy.go

```diff
@@ -174,23 +174,53 @@ func (h *UpgradeAwareProxyHandler) tryUpgrade(w http.ResponseWriter, req *http.R
 func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 	dialAddr := netutil.CanonicalAddr(h.Location)
 
+	var dialer func(network, addr string) (net.Conn, error)
```

Comment:
hrm, I'm sorry to be so pedantic about this, but how can tls.Dial take a nil config? does it set ServerAddress based on dialAddr?